src/common/commands_parser.rs

```diff
@@ -274,6 +280,137 @@ impl PcrArgs {
     }
 }
 
+/// The arguments used by `sign-eif` command
+#[derive(Debug, Clone)]
+pub struct SignArgs {
+    /// The method used for signing the EIF
+    pub signing_method: String,
+    /// The path to the enclave image file.
+    pub eif_path: String,
+    /// The path to the signing certificate.
+    pub signing_certificate: String,
+    /// The path to the private key.
+    pub private_key: Option<String>,
+    /// The region in which the KMS key resides.
+    pub region: Option<String>,
+    /// The KMS key id.
+    pub key_id: Option<String>,
+}
+
+impl SignArgs {
+    /// Construct a new `SignArg` instance from the given command-line arguments.
+    pub fn new_with(args: &ArgMatches) -> NitroCliResult<Self> {
+        let signing_method = parse_signing_method(args)
+            .map_err(|err| err.add_subaction("Parse signing method".to_string()))?;
+        let private_key = parse_private_key(args);
+        let region = parse_region(args);
+        let key_id = parse_key_id(args);
```

Comment:
`key_id` is a bit hard to interpret as "KMS Key ID". Perhaps `kms_key_arn` make more sense. Also `region` is not very obvious.

Reply:
`region` should also be optional. `kms_client` should load it from `.aws/config` or from `AWS_REGION` env variable. Perhaps there's some way to check if any of these are set or not.